src/js/utils/css.js

```diff
@@ -101,33 +101,41 @@ define([
         });
     };
 
-    var hexToRgba = function (hexColor, opacity) {
+    var canvasColorContext;
+    var getRgba = function (color, opacity) {
         var style = 'rgb';
-        if (hexColor) {
-            hexColor = String(hexColor).replace('#', '');
-            if (hexColor.length === 3) {
-                hexColor = hexColor[0] + hexColor[0] + hexColor[1] + hexColor[1] + hexColor[2] + hexColor[2];
-            }
-        } else {
-            hexColor = '000000';
-        }
-        var channels = [
-            parseInt(hexColor.substr(0, 2), 16),
-            parseInt(hexColor.substr(2, 2), 16),
-            parseInt(hexColor.substr(4, 2), 16)
-        ];
-        if (opacity !== undefined && opacity !== 100) {
+        var hasAlpha = (opacity !== undefined && opacity !== 100);
+        if (hasAlpha) {
             style += 'a';
-            channels.push(opacity / 100);
         }
-        return style + '(' + channels.join(',') + ')';
+        if (!canvasColorContext) {
```

Comment:
When doing timeline profiling of view setup, I've never seen a point where this code stood out. Moving out of the view and updating the util removes code that duplicates functionality and ensures that we only need this one context for all players on the page.